src/edu/macalester/comp124/hw3/PhotoWall.java

```diff
@@ -0,0 +1,187 @@
+     * @param ypo y position of the picture
+     * @param filepath where to locate the picture
+     * @param faceX x position of the face object
+     * @param faceY y position of the face object
+     * @param faceW width of the face object
+     * @param faceH height of the object
+     */
+    private void addRectangleFrame(double x, double y, double width, double height,float frameWidth, Color color,int xpo, int ypo, String filepath,
+                                   double faceX,double faceY,double faceW,double faceH){
+        RectangleFramePicture framePicture=new RectangleFramePicture(x,y,width,height);
+        framePicture.drawFrame(frameWidth,color);
+        this.add(framePicture);
+        framePicture.addPicture(xpo,ypo,filepath);
+        framePicture.addFlower(faceX,faceY,faceW,faceH);
+    }
+
+    /**
+     * add ovalframe to photowall
+     * @param x x position of the frame
+     * @param y y position of the frame
+     * @param width width of of the oval
+     * @param height height of the oval
+     * @param frameWidth width of the frame
+     * @param color color of the frame
+     * @param xpo x position of the picture
+     * @param ypo y position of the picture
+     * @param filepath where to locate the picture
+     * @param faceX x position of the face object
+     * @param faceY y position of the face object
+     * @param faceW width of the face object
+     * @param faceH height of the object
+     */
+    private void addOvalFrame(double x, double y, double width, double height,float frameWidth, Color color,int xpo, int ypo, String filepath,
+                              double faceX,double faceY,double faceW,double faceH){
+        OvalFramePicture framePicture=new OvalFramePicture(x,y,width,height);
+        framePicture.drawFrame(frameWidth,color);
+        this.add(framePicture);
+        framePicture.addPicture(xpo,ypo,filepath);
+        framePicture.addFace(faceX,faceY,faceW,faceH);
+    }
+
+    private void addRandomRectangleFrame(double x,double y){
+        RectangleFramePicture frame=new RectangleFramePicture(x,y,300,200);
+        frame.drawFrame(200,Color.green);
+        this.add(frame);
+        Random read=new Random();
+        int check=read.nextInt(2);
+        if (check==1){
+            frame.addFlower(20,20,50,50);
+        }else if(check==0){
+            frame.addFace(20,20,50,50);
+        }
+
+
+
+    }
+
+    private void addRandomOvalFrame(double x,double y){
+        OvalFramePicture frame=new OvalFramePicture(x,y,300,200);
+        frame.drawFrame(200,Color.red);
+        this.add(frame);
+        Random read=new Random();
+        int check=read.nextInt(2);
+        if (check==1){
+            frame.addFlower(50,30,50,50);
+        }else if(check==0){
+            frame.addFace(20,20,50,50);
+        }
+
+
+
+    }
+
+
+    @Override
+    public void mouseClicked(MouseEvent e) {
+        Point location=e.getPoint();
+        double x = location.getX();
```

Comment:
You need to center the image at the point where the mouse clicked - it just involves a little math with the width/height of the image and where the mouse was clicked to put it at the center instead of the top left (-1)